File: video-streaming-frontend/src/app/dto/video.dto.ts
import { CommentDto } from './comment.dto';
import { UserDto } from './user.dto';

export interface VideoDto {
  id: number;
  description: string;
  title: string;
  userId: number;
  likes: UserDto[];
  dislikes: UserDto[];
  tags: string[];
  videoUrl: string;
  videoStatus: string;
  views: UserDto[];
  thumbnailUrl: string;
  comments: CommentDto[];
}

File: video-streaming-backend/user-service/src/main/java/com/fourbit/sachintha/dto/VideoDto.java
package com.fourbit.sachintha.dto;

import java.util.ArrayList;
import java.util.List;

import com.fourbit.sachintha.model.VideoStatus;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class VideoDto {
  private Long id;
  private String description;
  private String title;
  private UserDto user;
  private String videoUrl;
  private VideoStatus videoStatus;
  private String thumbnailUrl;
  private List<String> tags = new ArrayList<>();
  private Integer likesCount = 0;
  private Integer dislikesCount = 0;
  private Long viewsCount = Long.valueOf(0);
  // private List<CommentDto> comments = new ArrayList<>();
  
}
